cleverhans/attacks_tf.py

```diff
@@ -417,3 +417,245 @@ def jacobian_augmentation(sess, x, X_sub_prev, Y_sub, grads, lmbda,
 
     # Return augmented training data (needs to be labeled afterwards)
     return X_sub
+
+
+class CarliniWagnerL2:
```

Comment:
I will kill this class. It's only a class because it was convenient when I was doing my own caching in the attack, but with the generate_np rewrite it should just be its own method.